### PR TITLE
Update create_functions.sql

### PR DESCRIPTION
**Pull request by dmdesimone**
For, UHM-4651, create a function to return primary identifier for a patient.

### CODE REVIEW

**Comment by mseaton** [comment]
@dmdesimone - I have another pending change to this file to look up global properties and user properties.  I think we should also add a function to get metadata mappings, which is currently embedded in this particular function.  Let me look at adding that in before we merge this...

File: api/src/main/resources/sql/procedures/create_functions.sql [hunk 1]
```diff
@@ -172,6 +172,36 @@ RETURN dosId_out;
 
 END;
 #
+
+/*
+The following will return the primary patient identifier defined in the system's global properties for a given patient id
```
Comment (mogoodrich) [inline api/src/main/resources/sql/procedures/create_functions.sql:177]: change to "defined in the system's metadata mappings"... sorry I led you astray with the GP solution but that has been deprecated.

File: api/src/main/resources/sql/procedures/create_functions.sql [hunk 2]
```diff
@@ -172,6 +172,36 @@ RETURN dosId_out;
...
+
+BEGIN
+    DECLARE  id VARCHAR(255);
+
+    SELECT pid.identifier into id from patient_identifier pid 
+    INNER JOIN metadatamapping_metadata_term_mapping m on m.code='emr.primaryIdentifierType'
+    where voided = 0 
+    and pid.identifier_type = 
+      (select pid2.patient_identifier_type_id from patient_identifier_type pid2 where
+      pid2.uuid = m.metadata_uuid)
```
Comment (mogoodrich) [inline api/src/main/resources/sql/procedures/create_functions.sql:196]: Not sure why we are doing the join in metadatamapping_metadata_term_mapping... can't we just do something like ...pid2.uuid = (select uuid from  metadatamapping_metadata_term_mapping where code='emr.primaryIdentifierType')

**Comment by mogoodrich** [comment]
@mseaton @dmdesimone I noticed this PR is still open?  Did we resolve this?  FWIW, create_functions has been moved to config-pihemr

**Comment by brandones** [comment]
Seems like this should be re-opened against config-pihemr if needed.

**Pull request closed without merging**